runtime/expr/cast.go

```diff
@@ -106,7 +106,7 @@ func (c *casterFloat16) Eval(ectx Context, val *zed.Value) *zed.Value {
 	if !ok {
 		return c.zctx.WrapError("cannot cast to float16", val)
 	}
-	return ectx.NewValue(zed.TypeFloat16, zed.EncodeFloat16(float32(f)))
+	return ectx.CopyValue(zed.NewFloat16(float32(f)))
```

Comment:
Why are we copying the value here if zed.NewFloat16 creates a new Value?

Reply:
So it'll be allocated by ectx rather than on the heap. This doesn't work the way we want right now because Context is an interface, and that cause its argument to escape to the heap. But I'm planning to fix that by converting Context from an interface to a struct.